src/views/exportJarFile.ts

```diff
@@ -0,0 +1,335 @@
+                                step = SOLVE_MAINMETHOD;
+                                break;
+                            }
+                            case SOLVE_MAINMETHOD: {
+                                pickResult = await this.resolveMainMethod(progress, token, pickSteps, projectFolder.uri.fsPath);
+                                step = GENERATE_JAR;
+                                break;
+                            }
+                            case GENERATE_JAR: {
+                                outputFileName = await this.generateJar(progress, token, pickSteps, rootNodes, pickResult, projectFolder.uri.fsPath);
+                                resolve(outputFileName);
+                                step = FINISH;
+                                break;
+                            }
+                        }
+                    } catch (err) {
+                        if (err === InputFlowAction.back) {
+                            step = pickSteps.pop();
+                            continue;
+                        } else {
+                            return reject();
+                        }
+                    }
+                }
+            });
+        }).then((message) => { this.successMessage(message); }, () => {});
+    }
+
+    private static resolveProject(progress, token: CancellationToken, pickSteps: string[], node?: INodeData): Promise<WorkspaceFolder> {
+        return new Promise<WorkspaceFolder>((resolve, reject) => {
+            if (token.isCancellationRequested) {
+                return reject();
+            }
+            const folders = workspace.workspaceFolders;
+            let projectFolder: WorkspaceFolder;
+            if (node instanceof WorkspaceNode) {
+                folders.forEach((folder) => {
+                    if (folder.uri.toString() === node.uri) {
+                        return resolve(folder);
+                    }
+                });
+                return reject();
+            }
+            if (folders && folders.length) {
+                if (folders.length === 1) {
+                    return resolve(folders[0]);
+                }
+                progress.report({ increment: 10, message: "Selecting project..." });
+                const pickNodes: QuickPickNode[] = [];
+                for (const folder of folders) {
+                    pickNodes.push(new QuickPickNode(folder.name, folder.uri.fsPath, folder.uri.fsPath));
+                }
+                const pickBox = window.createQuickPick<QuickPickNode>();
+                pickBox.items = pickNodes;
+                pickBox.title = "Export Jar - Determine project";
+                pickBox.placeholder = "Select the project...";
+                pickBox.ignoreFocusOut = true;
+                pickBox.onDidAccept(() => {
+                    pickSteps.push(SOLVE_PROJECT);
+                    folders.forEach((folder) => {
+                        if (folder.uri.fsPath === pickBox.selectedItems[0].uri) {
```

Comment:
when will the `!==` happens?

Reply:
The chosen folder's uri is `pickBox.selectedItems[0].uri` and this block is to find out the chosen one among workspace folders. `!==` may happen when the chosen folder is not in front of the folders list.